kotskinds/apis/kots/v1beta1/helmchart_types.go

```diff
@@ -174,6 +174,57 @@ func (h *HelmChartSpec) GetHelmValues(values map[string]MappedChartValue) (map[s
 	return result, nil
 }
 
+func MergeHelmChartValues(baseValues map[string]MappedChartValue,
+	overlayValues map[string]MappedChartValue) map[string]MappedChartValue {
+
+	result := map[string]MappedChartValue{}
+	for k, v := range baseValues {
+		if _, exists := overlayValues[k]; !exists {
+			result[k] = baseValues[k]
+			continue
+		}
+		if v.valueType != "children" {
+			result[k] = overlayValues[k]
+		} else {
+			result[k] = MappedChartValue{
+				valueType: "children",
+				children:  mergeValueChildren(v.children, overlayValues[k].children),
+			}
+		}
+	}
+	for k, v := range overlayValues {
+		if _, exists := baseValues[k]; !exists {
+			result[k] = v
+		}
+	}
+	return result
+}
+
+func mergeValueChildren(baseValues map[string]*MappedChartValue, overlayValues map[string]*MappedChartValue) map[string]*MappedChartValue {
```

Comment:
is this the same (or almost the same) as the function above? if so why duplicate?

Reply:
yes i can explain, the top level struct passed is map[string]MappedChartValue, but the children is map[string]*MappedChartValue.  unless have two functions, it was not possible to recurse.